sematic/ee/plugins/tracked_plugin_mixin.py

```diff
@@ -0,0 +1,35 @@
+# Standard Library
+import logging
+
+# Third-party
+import posthog  # type: ignore
+
+# Sematic
+from sematic.db.models.user import User
+
+logger = logging.getLogger(__name__)
+
+
+class TrackedPluginMixin:
+    """
+    A mixin for basic usage tracking of EE plug-ins.
+    """
+
+    @classmethod
+    def track_usage(cls, **kwargs) -> None:
+        """
+        Register a tracking event.
+        """
+        user_id = None
+        if "user" in kwargs:
+            user: User = kwargs["user"]
+            user_id = user.email
+
+        event = kwargs.get("event", "used")
+
+        posthog.project_api_key = "phc_zwWXIEnEuZNj9JxC870IrxocjcuxG5ZjNoaNdW8YgXF"
```

Comment:
This should probably be a constant somewhere in EE

Reply:
I was thinking making it an explicit constant would make it even easier to steal, but that's arbitrary. One way to add a barrier would be to store it in a file in a different encoding (e.g. base64). Does not prevent anyone from decoding it, but adds additional work.

Reply:
I don't know that there's much risk in anybody stealing it anyway, aside from skewing our numbers or forcing us to bump against API quotas.